sdk/storage/Azure.Storage.Files.Shares/perf/Azure.Storage.Files.Shares.Perf/Scenarios/DownloadFile.cs

```diff
@@ -0,0 +1,151 @@
+        /// When multiple instances are run in parallel, the setup will take place once, prior to the
+        /// execution of the first test instance.
+        /// </summary>
+        public override async Task GlobalSetupAsync()
+        {
+            await base.GlobalSetupAsync();
+
+            await _fileUploader.GlobalSetupAsync();
+        }
+
+        /// <summary>
+        /// Performs the tasks needed to clean up the environment for the test scenario.
+        /// When multiple instances are run in parallel, the cleanup will take place once,
+        /// after the execution of all test instances.
+        /// </summary>
+        public override async Task GlobalCleanupAsync()
+        {
+            await base.GlobalCleanupAsync();
+
+            await _fileUploader.GlobalCleanupAsync();
+        }
+
+        /// <summary>
+        /// Performs the tasks needed to initialize and set up the environment for an instance of the test scenario.
+        /// When multiple instances are run in parallel, setup will be run once for each prior to its execution.
+        /// </summary>
+        public override async Task SetupAsync()
+        {
+            await base.SetupAsync();
+
+            // Upload a file to the share storage. The test will use 'ShareClient' to download this file.
+            (FilesShareClient, DirectoryClientName, FileClientName) = await _fileUploader.UploadARandomFileAsync();
+        }
+
+        /// <summary>
+        /// Performs the tasks needed to clean up the environment for an instance of the test scenario.
+        /// When multiple instances are run in parallel, cleanup will be run once for each after execution has completed.
+        /// </summary>
+        public override async Task CleanupAsync()
+        {
+            await base.CleanupAsync();
+
+            try
+            {
+                await _fileUploader.CleanupAsync();
+                File.Delete(DownloadFilePath);
+            }
+            catch
+            {
+            }
+        }
+
+        /// <summary>
+        /// Executes the performance test scenario synchronously.
+        /// </summary>
+        /// <param name="cancellationToken">The token used to signal when cancellation is requested.</param>
+        public override void Run(CancellationToken cancellationToken)
+        {
+            ShareDirectoryClient directoryClient = FilesShareClient.GetDirectoryClient(DirectoryClientName);
+            ShareFileClient fileClient = directoryClient.GetFileClient(FileClientName);
+
+            Models.ShareFileDownloadInfo fileDownloadInfo = fileClient.Download(cancellationToken: cancellationToken);
+
+            // We only want to measure the download operation. So, skip the part of creating a local file with the downloaded stream.
+            //using (FileStream stream = File.OpenWrite(DownloadFilePath))
+            //{
+            //    fileDownloadInfo.Content.CopyTo(stream);
+            //}
```

Comment:
This stream needs to be read to completion, otherwise you're just testing the latency of starting a download.

Reply:
oh there is another thread about this in the async overload. Will move this there.